eo-maven-plugin/src/main/java/org/eolang/maven/optimization/OptSpy.java

```diff
@@ -42,11 +42,26 @@ public final class OptSpy implements Optimization {
     private final Path target;
 
     /**
-     * The main constructor.
-     * @param target Where to track optimization steps.
+     * Apply versions shifts or not.
+     */
+    private final boolean versioned;
```

Comment:
@maxonfjvipon The solution with the `versioned` field is highly questionable. First, it leads to hidden [configurable objects](https://www.yegor256.com/2016/04/19/object-must-not-be-configurable.html). Secondly, it necessitates numerous changes in various locations, which is a bad sign. Might there be an alternative solution? Can we apply `move-versions-deeper.xsl` to all parsing pipelines (independently if we have versions or not)?